clientconn.go

```diff
@@ -71,6 +71,18 @@ type dialOptions struct {
 // DialOption configures how we set up the connection.
 type DialOption func(*dialOptions)
 
+// NameResolver defines the functions needed to resolve a customized names.
+type NameResolver interface {
+	// Get returns the map of customized names and the corresponding ip/DNS addresses.
+	Get(target string) map[string]string
+	// Watch set a watcher on the target directory/key.
+	Watch(target string)
+	// GetUpdate returns the key and value of the updated namePair of the target directory/key being watched.
+	GetUpdate() (string, string)
```

Comment:
GetUpdate returns a name resolution change when watch is triggered. It blocks until it observes a change. The caller needs to call it again to get the next change.